[HumanBenchmark][Frontend][Header] implement the signin/signup modal

Benjamin-Snyder opened the pull request:
I implemented the login and sign up modal to work with their associated buttons in the header

jasonthaisy commented on front-end/src/Header/Header.jsx:
```diff
@@ -1,8 +1,29 @@
+import {useState} from "react";
 
 const Header = ({ isLoggedIn, setIsLoggedIn }) => {
+  const [showSignInModal, setShowSignInModal] = useState(false);
+  const [showSignUpModal, setShowSignUpModal] = useState(false);
+
+  const handleSignInClick = () => {
+    setShowSignInModal(true);
+    setShowSignUpModal(false); // only one modal can be open at a time
+  };
+
+  const handleSignUpClick = () => {
+    setShowSignUpModal(true);
+    setShowSignInModal(false); // only one modal can be open at a time
+  };
+
+  const handleCloseModals = () => {
+    setShowSignInModal(false);
+    setShowSignUpModal(false);
+  };
+
```
Also update the screen recording once this is done

Benjamin-Snyder commented:
Addressed comments:
created a single state for which modal is shown

Video shows it still works

https://github.com/user-attachments/assets/e23d59bc-8218-48de-9c1f-087b44e0c16f
